Update constructors (#57)
JuliaSmoothOptimizers/BundleAdjustmentModels.jl: README.md

   2 │ problem-73-11032-pre  ladybug  92244  33753  1106928
```

> You can get the name of the problem by handling the dataframe.

Reviewer: ```suggestion
You can get the problem name directly from the dataframe:
```

You can directly construct a nonlinear least-squares model based on [NLPModels](http://juliasmoothoptimizers.github.io/NLPModels.jl/latest/):

```julia
julia> model = BundleAdjustmentModel("problem-49-7776-pre")
BundleAdjustmentModel{Float64, Vector{Float64}}
```

You can also construct a nonlinear least-squares model by giving the constructor the path to the archive.

> You must also set `direct_path = true`

Reviewer: What does "direct path" mean?

Author: "direct_path" means you directly give the path to a problem file instead of using the system of artifacts.
It should almost never be the case but I left it in case someone wants to modify some of the problems themselves.

Reviewer: I don't think we want to encourage that.

Author: I will delete it in this case.

You can directly construct a nonlinear least-squares model based on [NLPModels](http://juliasmoothoptimizers.github.io/NLPModels.jl/latest/):

```julia
julia> model = BundleAdjustmentModel("problem-49-7776")
BundleAdjustmentModel{Float64, Vector{Float64}}
```

You can then evaluate the residual and jacobian (or their in-place version) from NLPModels:

```julia
julia> using NLPModels
```

```julia
julia> residual(model, model.meta.x0)
63686-element Vector{Float64}:
 -9.020226301243213
 11.263958304987227
  ⋮
 -0.01443314653496941
 -0.4486499211288866
```

```julia
julia> meta_nls = nls_meta(model)

> julia> S = typeof(model.meta.x0)

Author: I corrected it in the latest commit.

> julia> Fx = S(undef, meta_nls.nequ)

Author: I corrected it in the latest commit.

 -0.4486499211288866
```

```julia
julia> meta_nls = nls_meta(model)
julia> S = typeof(model.meta.x0)
julia> F = S(undef, meta_nls.nequ)
julia> residual!(model, model.meta.x0, F)
63686-element Vector{Float64}:
 -9.020226301243213
 11.263958304987227
  ⋮
 -0.01443314653496941
 -0.4486499211288866
```

> You need to call `jac_structure_residual!` at least once before calling `jac_op_residual`.

Reviewer: why?

Author: The `jac_op_residual` I use is based on the sparse matrix so I need to create it before using it because it doens't do it himself.

Reviewer: But I'm confused because when you call `jac_op_residual` below (on line 157), you don't use `rows`, `cols` or `vals`.

Author: You are right, and I didn't even test if the last `jac_op_residual` was wroking correctly. I will digger into this to see if it works.

```julia
julia> S = typeof(model.meta.x0)
julia> meta_nls = nls_meta(model)
julia> rows = Vector{Int}(undef, meta_nls.nnzj)
julia> cols = Vector{Int}(undef, meta_nls.nnzj)
julia> jac_structure_residual!(model, rows, cols)
([1, 1 … 63686, 63686], [1, 2 … 23768, 23769])
```

> You need to call `jac_coord_residual!` everytime before calling `jac_op_residual`.

Reviewer: why?

Author: The `jac_op_residual` function doesn't update the matrix so it needs to be done manually at each iteration.

 -0.4486499211288866
```

You need to call `jac_structure_residual!` at least once before calling `jac_op_residual!`.

```julia
julia> meta_nls = nls_meta(model)
julia> rows = Vector{Int}(undef, meta_nls.nnzj)
julia> cols = Vector{Int}(undef, meta_nls.nnzj)
julia> jac_structure_residual!(model, rows, cols)
([1, 1 … 63686, 63686], [1, 2 … 23768, 23769])
```

You need to call `jac_coord_residual!` everytime before calling `jac_op_residual!`.

```julia
julia> S = typeof(model.meta.x0)

> julia> vals = S(undef, meta_nls.nnzj)

Author: I corrected it in the latest commit.

```

```julia

> julia> Jv = S(undef, meta_nls.nequ)

Reviewer: use similar

Author: I corrected it in the latest commit.
